### PR TITLE
Add User profile icon to Textbook navigation

### LINKED ISSUE #796: User Account: Add a User Icon to the Textbook nav
Following the comment [here](https://github.com/Qiskit/platypus/issues/258#issuecomment-1061164366)

In order to finish the User Account user story, we'll need a user icon that will allow users to log in.

### PR DESCRIPTION
**Pull request by techtolentino**
## Changes
Closes #796 

## Implementation details
- added User icon button link to the nav
- when a user clicks the button, they're taken to their `/account` page
- (if the user isn't logged in, they'll hit the `/signin` page)
- added mobile `Profile` link to mobile-menu

## How to read this PR
- review updates to Textbook menu, + mobile-menu
- review style updates

## Screenshots
https://user-images.githubusercontent.com/6276074/157519554-ecf51561-cee0-4246-a098-c398e523fb3f.mov

### CODE REVIEW

**Comment by techtolentino** [comment]
I didn't consider this earlier, but I'm converting this to draft until the [CourseList bug](https://github.com/Qiskit/platypus/issues/783) is resolved

File: frontend/vue/components/TheMenu/MobileMenu.vue [hunk 1]
```diff
@@ -72,7 +82,7 @@ class Props {
 
 @Options({
   components: {
-    AppLink
+    AppLink, BasicLink, User16
```
Comment (korgan00) [inline frontend/vue/components/TheMenu/MobileMenu.vue:85]: We use to have one component per line

File: frontend/vue/components/TheMenu/MobileMenu.vue [hunk 2]
```diff
@@ -105,6 +116,8 @@ export default class MobileMenu extends mixins(MenuMixin) {
 @import '~/../scss/variables/mq.scss';
 @import '~/../scss/mixins/mixins.scss';
 
+$mobile-menu-item-height: 4rem;
```
Comment (korgan00) [inline frontend/vue/components/TheMenu/MobileMenu.vue:119]: I like this.

File: frontend/vue/components/TheMenu/MobileMenu.vue [hunk 3]
```diff
@@ -181,6 +194,23 @@ export default class MobileMenu extends mixins(MenuMixin) {
...
 
+  &__account-link {
+    display: flex;
+    align-items: center;
+    padding-left: $spacing-07;
+    padding-right: $spacing-05;
+    height: $mobile-menu-item-height;
+    border-bottom: 1px solid $border-color-light;
+    &__icon {
+      border: 1px solid $carbon--purple-70;
```
Comment (korgan00) [inline frontend/vue/components/TheMenu/MobileMenu.vue:205]: I think `$border-color-secondary` is better here

File: frontend/vue/components/TheMenu/MobileMenu.vue [hunk 4]
```diff
@@ -181,6 +194,23 @@ export default class MobileMenu extends mixins(MenuMixin) {
...
+      border-radius: 50%;
+      height: 1.5rem;
+      width: 1.5rem;
+      margin-right: $spacing-03;
+      fill: $carbon--purple-70;
```
Comment (korgan00) [inline frontend/vue/components/TheMenu/MobileMenu.vue:210]: Here we should use also a variable defined by us

File: frontend/vue/components/TheMenu/index.vue [hunk 5]
```diff
@@ -105,9 +113,10 @@ import MobileMenu from './MobileMenu.vue'
 
 // using Carbon web-component from https://github.com/carbon-design-system/carbon-web-components#basic-usage
 import 'carbon-web-components/es/components/dropdown/dropdown.js'
+import 'carbon-web-components/es/components/button/button.js'
 
 @Options({
-  components: { AppLink, AppLogo, MobileMenu, BasicLink, Menu20, Close20 }
+  components: { AppLink, AppLogo, MobileMenu, BasicLink, Menu20, Close20, User20 }
```
Comment (korgan00) [inline frontend/vue/components/TheMenu/index.vue:119]: We use to have one component per line

File: frontend/vue/components/TheMenu/index.vue [hunk 6]
```diff
@@ -164,6 +172,33 @@ export default class TheMenu extends mixins(MenuMixin) {
...
+      transition: background-color 70ms cubic-bezier(0, 0, 0.38, 0.9) 0s;
+      padding-right: 15px;
+
+      &:focus {
+        border-color: $carbon--purple-70;
+        box-shadow: initial;
+      }
+
+      &:hover {
+        background-color: $carbon--purple-80;
```
Comment (korgan00) [inline frontend/vue/components/TheMenu/index.vue:197]: We should use the colors in the scss.

**Review by korgan00 — CHANGES_REQUESTED** [review]

**Comment by korgan00** [comment]
Good job Randy!
Just few small and quick changes

**Review by korgan00 — APPROVED** [review]
LGTM. Thank you!

File: frontend/vue/components/TheMenu/MobileMenu.vue [hunk 7]
```diff
@@ -60,8 +68,10 @@
 <script lang="ts">
 // import { Component, Mixins, Prop, Watch } from 'vue-property-decorator'
```
Comment (eddybrando) [inline frontend/vue/components/TheMenu/MobileMenu.vue:69]:
```suggestion
```

**Review by eddybrando — APPROVED** [review]
This is great, @techtolentino! Thank you very much!

**Pull request merged**